nextcord/client.py

```diff
@@ -2023,33 +2039,171 @@ def get_application_command(self, command_id: int) -> Optional[BaseApplicationCo
+        )
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        *,
+        type: Union[int, ApplicationCommandType],
+        qualified_name: str,
+        guild_id: Optional[int],
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        qualified_name: str,
+        type: Union[int, ApplicationCommandType],
+        guild_id: Optional[int],
+        *,
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        type: Union[int, ApplicationCommandType],
+        qualified_name: str,
+        guild_id: Optional[int],
+        *,
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @copy_doc(__actual_get_application_command_from_signature)
+    def get_application_command_from_signature(
+        self, *args: Any, search_locales: bool = False, **kwargs: Any
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        REQUIRED_ARGS = ("qualified_name", "type", "guild_id")
+        FUNC = "get_application_command_from_signature()"
+        ARGS_DEPRECATED_MESSAGE = "The signature passed to {name} is deprecated, please use keyword arguments only instead. Converted signature: {converted_signature}."
+        KWARG_DEPRECATION_MESSAGE = 'The keyword argument "{old}=" used in {name} is deprecated, please use "{new}=" instead.'
+
+        kwargs_convert_table: Dict[str, Any] = {"name": "qualified_name", "cmd_type": "type"}
+        VALID_KWARGS = ("type", "qualified_name", "guild_id", "search_locales")
+        for kw in kwargs.copy():
+            if kw in kwargs_convert_table:
+                warnings.warn(
+                    KWARG_DEPRECATION_MESSAGE.format(
+                        name=FUNC, old=kw, new=kwargs_convert_table[kw]
+                    ),
+                    stacklevel=2,
+                    category=FutureWarning,
+                )
+                kwargs[kwargs_convert_table[kw]] = kwargs.pop(kw)
+            elif kw not in VALID_KWARGS:
+                raise TypeError(f"{FUNC} got an unexpected keyword argument '{kw}'")
+
+        if len(args) > 3:
+            raise TypeError(f"{FUNC} takes 3 positional argument but {len(args)} were given")
+        for index, arg in enumerate(args):
+            if index in (0, 1):
+                if isinstance(arg, str):
+                    kwargs["qualified_name"] = kwargs.get("qualified_name", arg)
+                elif isinstance(arg, (int, ApplicationCommandType)):
+                    kwargs["type"] = kwargs.get("type", arg)
+            elif index == 2:
+                kwargs["guild_id"] = kwargs.get("guild_id", arg)
```

Comment:
Shouldn't the type of the argument also be checked here to be in line with the other if statements?

Reply:
No, the others are changed because of the original order, this should always be int or None.